src/model.c

```diff
@@ -344,40 +345,42 @@ size_t RAI_ModelRunCtxNumInputs(RAI_ModelRunCtx* mctx) {
     return 0;
   }
   // Here we assume batch is well-formed (i.e. number of outputs is equal in all batches)
-  return array_len(mctx->batches[0].inputs);
+  return mctx->batches[0].ninputs;
 }
 
 size_t RAI_ModelRunCtxNumOutputs(RAI_ModelRunCtx* mctx) {
-  if (RAI_ModelRunCtxNumBatches(mctx) == 0) {
+  if (RAI_ModelRunCtxNumBatches(mctx)) {
```

Comment:
hey @filipecosta90 are you sure about this?

Reply:
I closed this one @lantiga . I was focusing on model.c instead of redisai.c. 
I've opened #321 precisely for that :)